core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java

```diff
@@ -217,9 +218,9 @@ public void refreshMapping(final String index, final String indexUUID) {
                     // failures here mean something is broken with our cluster state - fail all tasks by letting exceptions bubble up
                     for (String index : request.indices()) {
                         final IndexMetaData indexMetaData = currentState.metaData().index(index);
-                        if (indexMetaData != null  && indicesService.hasIndex(index) == false) {
+                        if (indexMetaData != null  && indicesService.hasIndex(indexMetaData.getIndex()) == false) {
```

Comment:
this the part I meant about skipping indices that do not exist anymore. We can accumulate a List<Index> todoList here and pass that to applyRequest, so we don't need to do another resolving round there.